scripts/composer/AssetInstaller.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @file
+ * Contains \DrupalProject\composer\AssetInstaller.
+ */
+
+namespace DrupalProject\composer;
+
+use Composer\Script\Event;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+class AssetInstaller {
+
+  public static function install(Event $event) {
+    $extra = $event->getComposer()->getPackage()->getExtra();
+    if (!isset($extra['drupal-app-dir'])) {
+      throw new \RuntimeException('Please configure drupal-app-dir in your composer.json');
+    }
+    if (!isset($extra['drupal-web-dir'])) {
+      throw new \RuntimeException('Please configure drupal-web-dir in your composer.json');
+    }
+
+    $fs = new Filesystem();
+    $fs->mkdir([$extra['drupal-app-dir'], $extra['drupal-web-dir']]);
+
+    static::createSymlinks($extra['drupal-app-dir'], $extra['drupal-web-dir']);
+
+    static::createStubPhpFile($extra['drupal-app-dir'], $extra['drupal-web-dir'], 'index.php');
+    static::createStubPhpFile($extra['drupal-app-dir'], $extra['drupal-web-dir'], 'core/install.php');
+    static::createStubPhpFile($extra['drupal-app-dir'], $extra['drupal-web-dir'], 'core/rebuild.php');
+
+    // Optional?
+    static::createStubPhpFile($extra['drupal-app-dir'], $extra['drupal-web-dir'], 'core/modules/statistics/statistics.php');
+
+    // Symlink public files
+    $fs = new Filesystem();
+    $fs->symlink(realpath($extra['drupal-app-dir']) . '/sites/default/files', $extra['drupal-web-dir'] . '/sites/default/files');
```

Comment:
IMO is a good thing create the files symlink using relative paths, absolute symbolic links don't work when the filesystem is mounted elsewhere.
`$fs->symlink('../../../' . $extra['drupal-web-dir'] . '/sites/default/files', $extra['drupal-app-dir'] . '/sites/default/files');`